### ISSUE TITLE
Running sqlplus scriptrunner step inside docker container 

### ISSUE THREAD

**Issue by raducanuoctavian** [jenkinsci/sqlplus-script-runner-plugin#70]
### Jenkins and plugins versions report

<details>
  <summary>Environment</summary>

  
  ```text
Jenkins: 2.249.2
OS: Linux - 3.10.0-693.21.1.el7.x86_64
---
JDK_Parameter_Plugin:1.0
ace-editor:1.1
active-directory:2.22
analysis-core:1.96
analysis-model-api:9.3.2
android-emulator:3.1.2
ant:1.11
antisamy-markup-formatter:2.1
apache-httpcomponents-client-4-api:4.5.10-2.0
authentication-tokens:1.4
bitbucket:1.1.27
bitbucket-pullrequest-builder:1.5.0
bitbucket-scm-filter-aged-refs:0.2.0
blueocean:1.24.3
blueocean-autofavorite:1.2.4
blueocean-bitbucket-pipeline:1.24.3
blueocean-commons:1.24.3
blueocean-config:1.24.3
blueocean-core-js:1.24.3
blueocean-dashboard:1.24.3
blueocean-display-url:2.4.0
blueocean-events:1.24.3
blueocean-executor-info:1.24.3
blueocean-git-pipeline:1.24.3
blueocean-github-pipeline:1.24.3
blueocean-i18n:1.24.3
blueocean-jira:1.24.3
blueocean-jwt:1.24.3
blueocean-personalization:1.24.3
blueocean-pipeline-api-impl:1.24.3
blueocean-pipeline-editor:1.24.3
blueocean-pipeline-scm-api:1.24.3
blueocean-rest:1.24.3
blueocean-rest-impl:1.24.3
blueocean-web:1.24.3
bootstrap4-api:4.5.3-1
bouncycastle-api:2.18
branch-api:2.6.2
build-name-setter:2.1.0
build-timeout:1.20
build-user-vars-plugin:1.6
checks-api:1.3.0
ci-game:1.26
claim:2.17
cloudbees-bitbucket-branch-source:2.9.5
cloudbees-folder:6.15
cobertura:1.16
code-coverage-api:1.3.1
command-launcher:1.5
conditional-buildstep:1.4.0
config-file-provider:3.7.0
copyartifact:1.45.2
credentials:2.4.1
credentials-binding:1.24
dashboard-view:2.14
data-tables-api:1.10.21-3
dependency-check-jenkins-plugin:5.1.2
discard-old-build:1.05
disk-usage:0.28
display-url-api:2.3.4
docker-commons:1.17
docker-java-api:3.1.5.2
docker-plugin:1.2.3
docker-workflow:1.25
durable-task:1.35
echarts-api:4.9.0-2
email-ext:2.79
emotional-jenkins-plugin:1.2
envinject:2.3.0
envinject-api:1.7
external-monitor-job:1.7
favorite:2.3.2
file-operations:1.11
flexible-publish:0.15.2
font-awesome-api:5.15.1-1
forensics-api:0.7.0
git:4.4.5
git-client:3.5.1
git-server:1.9
github:1.32.0
github-api:1.117
github-branch-source:2.9.1
gradle:1.36
groovy:2.3
handlebars:1.1.1
handy-uri-templates-2-api:2.1.8-1.0
htmlpublisher:1.25
http_request:1.8.27
jackson2-api:2.12.1
jacoco:3.1.0
javadoc:1.6
jdk-tool:1.4
jenkins-design-language:1.24.3
jenkins-jira-plugin:3.6.0
jira:3.1.3
jira-steps:1.6.0
job-import-plugin:3.4
jquery:1.12.4-1
jquery-detached:1.2.1
jquery3-api:3.5.1-2
jsch:0.1.55.2
junit:1.44
ldap:1.26
ldapemail:0.8
lockable-resources:2.10
login-theme:1.1
mailer:1.32.1
mapdb-api:1.0.9.0
matrix-auth:2.6.4
matrix-groovy-execution-strategy:1.0.7
matrix-project:1.18
maven-plugin:3.8
mercurial:2.12
momentjs:1.1.1
multibranch-build-strategy-extension:1.0.10
multiple-scms:0.6
nexus-artifact-uploader:2.13
nodejs:1.3.9
nodelabelparameter:1.7.2
okhttp-api:3.14.9
pam-auth:1.6
parameterized-scheduler:1.0
parameterized-trigger:2.39
performance:3.18
pipeline-build-step:2.13
pipeline-github-lib:1.0
pipeline-graph-analysis:1.10
pipeline-input-step:2.12
pipeline-milestone-step:1.3.1
pipeline-model-api:1.7.2
pipeline-model-declarative-agent:1.1.1
pipeline-model-definition:1.7.2
pipeline-model-extensions:1.7.2
pipeline-rest-api:2.19
pipeline-stage-step:2.5
pipeline-stage-tags-metadata:1.7.2
pipeline-stage-view:2.19
pipeline-utility-steps:2.6.1
plain-credentials:1.7
plot:2.1.8
plugin-util-api:1.6.1
popper-api:1.16.0-7
port-allocator:1.8
promoted-builds:3.7
promoted-builds-simple:1.9
publish-over:0.22
publish-over-cifs:0.16
publish-over-ftp:1.15
pubsub-light:1.13
quality-gates:2.5
release:2.10.2
remote-file:1.11
resource-disposer:0.14
run-condition:1.3
schedule-build:0.5.1
scm-api:2.6.4
scm-filter-branch-pr:0.5.1
script-security:1.75
simple-theme-plugin:0.7
snakeyaml-api:1.27.0
sonar:2.12
sqlplus-script-runner:2.0.14
sse-gateway:1.24
ssh-agent:1.20
ssh-credentials:1.18.1
ssh-slaves:1.31.2
ssh-steps:2.0.0
statusmonitor:1.3
structs:1.23
subversion:2.13.2
text-finder:1.15
timestamper:1.11.8
token-macro:2.12
trilead-api:1.0.12
validating-string-parameter:2.5
variant:1.3
warnings-ng:8.4.4
windows-slaves:1.7
workflow-aggregator:2.6
workflow-api:2.40
workflow-basic-steps:2.22
workflow-cps:2.86
workflow-cps-global-lib:2.17
workflow-durable-task-step:2.37
workflow-job:2.40
workflow-multibranch:2.22
workflow-scm-step:2.11
workflow-step-api:2.23
workflow-support:3.6
ws-cleanup:0.38
  ```

</details>

### What Operating System are you using (both controller, and any agents involved in the problem)?

Docker container using the following image OS:

```
NAME="Ubuntu"
VERSION="18.04.6 LTS (Bionic Beaver)"
Docker version 1.13.1, build 7d71120/1.13.1
```

Docker is running on the following OS:
```
NAME="Ubuntu"
VERSION="18.04.4 LTS (Bionic Beaver)"
```

Location of sqlplus is the following : 
```
/abs/oracle/instantclient_19_9
```
**There is no tnsnames.ora or anything else written inside network/admin folder. It is a fresh instant client instance on the image.**

### Reproduction steps

Create a pipeline which has an sqlplus runner step inside of it:
```
stage("Sqlplus runner stage") {
  agent {
    docker {
	    image "oracle:image"
	    label "jenkins-slave"
    }
  }
  script {
	step([$class: 'SQLPlusRunnerBuilder',credentialsId:"${credentialsID}", instance:"${sid}", customOracleHome:"${env.ORACLE_HOME}", customSQLPlusHome:"${customSQLPlusHome}", scriptType:'userDefined', script: '', scriptContent: "spool ${logFileName}.log \n"+
					"${scriptcontent}\n"+
					"spool off"])
  }
}
```

### Expected Results

Running the script as it would outside of the container. I don't know if attaching a picture here would help with the cause, it's the basic console output of sqlplus script log.

### Actual Results

```
docker exec --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** 633bde8d22c120ffeb8149a4f9c03368b1ad68de9ffd360bafa4204c6d1b413b /abs/oracle/instantclient_19_9/sqlplus -v
```
The stage earlier mentioned returns a docker exec instead of just calling the sqlplus. 
The script exec jenkins log:
```
docker exec --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** 633bde8d22c120ffeb8149a4f9c03368b1ad68de9ffd360bafa4204c6d1b413b /abs/oracle/instantclient_19_9/sqlplus -L ******** @/abs/workspace/workspace12/temp-script-165719436740517576000697588527246.sql
```
Finally it returns:
```
14:46:11  ERROR:
14:46:11  ORA-12154: TNS:could not resolve the connect identifier specified
14:46:11  
14:46:11  
14:46:11  SP2-0751: Unable to connect to Oracle.  Exiting SQL*Plus
14:46:09  Process exited with status 1
```
I tried to following:
1. Tried to run a container and connect to it using the credentials used with jenkins - **worked**
2. Tried to run the sqlplus process just like sqlplus runner did, using docker exec (the top command) - **worked**

### Anything else?

I have to add the following connect method for sqlplus that we use:
```
sqlplus -L {db_schema_name}/{db_password}@{server_ip}:{port}/{sid}
```
In the earlier section I said I tried 2 methods of connecting with sqlplus. Both methods worked with the method mentioned earlier.
```sqlplus -L {db_schema_name}/{db_password}@{server_ip}:{port}/{sid}```

I can update the plugin version if this was fixed before, I tried finding if it was an old issue regarding the same thing but it seems that it's not the case.

**Comment by boaglio** [comment]
I don't see any bug, just a TNS error, there is something wrong with your env. 
You can also use _tnsping_ to test your connection.

**Comment by raducanuoctavian** [comment]
Hello @boaglio and thank you for your answer.
As I was saying in the post, testing it without jenkins seems to work.
The tests I conducted were the following:
1. Spin up a container and use 
```
sqlplus -L {db_schema_name}/{db_password}@{server_ip}:{port}/{sid} 
```
to see if the database is reachable. The result is that the database is **reachable**.
2. Try to run sqlplus with the command used by jenkins while running the step : 
```
docker exec --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** 633bde8d22c120ffeb8149a4f9c03368b1ad68de9ffd360bafa4204c6d1b413b /abs/oracle/instantclient_19_9/sqlplus -L ******** @/abs/workspace/workspace12/test-script.sql
```
Again, this method works, database was reched and everything runned ok.
The problem appears while running sqlplus from **inside jenkins**. 
I also conducted a test inside a sandbox with a step of the plugin on an agent with a docker container running. To be more precise and exclude the posiblity of wrong credentials - I runned the command without hiding the credentials from jenkins.
To be sure that the credentials I used were correct, I tested the command on my own computer beforehand. 
**The result was that it still does not work**.
I will add the following test in the sandbox that I conducted.

The console:

```
[2022-07-07T13:53:27.153Z] Running in Durability level: PERFORMANCE_OPTIMIZED
[2022-07-07T13:53:27.161Z] Loading library utilLibrary@master
[2022-07-07T13:53:27.162Z] Attempting to resolve master from remote references...
[2022-07-07T13:53:27.162Z]  > git --version # timeout=10
[2022-07-07T13:53:27.172Z]  > git --version # 'git version 2.11.0'
[2022-07-07T13:53:27.172Z] using GIT_ASKPASS to set credentials 
[2022-07-07T13:53:27.173Z]  > git ls-remote -h -- https://bitbucket.internal.domain.com/scm/devops/jenkins.git # timeout=10
[2022-07-07T13:53:27.369Z] Found match: refs/heads/master revision 89fd65963e8dea05ca2d36a4c3f45f92d35bede9
[2022-07-07T13:53:27.371Z] The recommended git tool is: NONE
[2022-07-07T13:53:27.371Z] using credential c872086e-0fcb-4a12-9971-e3e79bfebdfb
[2022-07-07T13:53:27.381Z]  > git rev-parse --is-inside-work-tree # timeout=10
[2022-07-07T13:53:27.393Z] Fetching changes from the remote Git repository
[2022-07-07T13:53:27.393Z]  > git config remote.origin.url https://bitbucket.internal.domain.com/scm/devops/jenkins.git # timeout=10
[2022-07-07T13:53:27.404Z] Fetching without tags
[2022-07-07T13:53:27.404Z] Fetching upstream changes from https://bitbucket.internal.domain.com/scm/devops/jenkins.git
[2022-07-07T13:53:27.404Z]  > git --version # timeout=10
[2022-07-07T13:53:27.414Z]  > git --version # 'git version 2.11.0'
[2022-07-07T13:53:27.415Z] using GIT_ASKPASS to set credentials 
[2022-07-07T13:53:27.415Z]  > git fetch --no-tags --progress -- https://bitbucket.internal.domain.com/scm/devops/jenkins.git +refs/heads/*:refs/remotes/origin/* # timeout=10
[2022-07-07T13:53:27.681Z] Checking out Revision 89fd65963e8dea05ca2d36a4c3f45f92d35bede9 (master)
[2022-07-07T13:53:27.681Z]  > git config core.sparsecheckout # timeout=10
[2022-07-07T13:53:27.694Z]  > git checkout -f 89fd65963e8dea05ca2d36a4c3f45f92d35bede9 # timeout=10
[2022-07-07T13:53:27.713Z]  > git rev-list --no-walk 89fd65963e8dea05ca2d36a4c3f45f92d35bede9 # timeout=10
[2022-07-07T13:53:27.791Z] [Pipeline] Start of Pipeline
[2022-07-07T13:53:28.118Z] [Pipeline] node
[2022-07-07T13:53:28.126Z] Running on Jenkins-NSIS-Linux2 in /abs/workspace/ORA Sandbox
[2022-07-07T13:53:28.130Z] [Pipeline] {
[2022-07-07T13:53:28.296Z] [Pipeline] isUnix
[2022-07-07T13:53:28.298Z] [Pipeline] sh
[2022-07-07T13:53:28.584Z] + docker inspect -f . nexus.internal.domain.com/docker-buildtools/builder:1.0
[2022-07-07T13:53:28.584Z] .
[2022-07-07T13:53:28.595Z] [Pipeline] withDockerContainer
[2022-07-07T13:53:28.678Z] Jenkins-NSIS-Linux2 does not seem to be running inside a container
[2022-07-07T13:53:31.245Z] $ docker run -t -d -u 1000:1000 -w "/abs/workspace/ORA Sandbox" -v "/abs/workspace/ORA Sandbox:/abs/workspace/ORA Sandbox:rw,z" -v "/abs/workspace/ORA Sandbox_tmp:/abs/workspace/ORA Sandbox_tmp:rw,z" -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** -e ******** nexus.internal.domain.com/docker-buildtools/builder:1.0 cat
[2022-07-07T13:53:31.910Z] $ docker top ee6c511036173bb15b790d0c952a4cdd0ed117ea5cfe9aec13a3ea96d168bf31 -eo pid,comm
[2022-07-07T13:53:29.474Z] [Pipeline] {
[2022-07-07T13:53:29.491Z] [Pipeline] stage
[2022-07-07T13:53:29.492Z] [Pipeline] { (Stage 1 - Checkout)
[2022-07-07T13:53:29.526Z] [Pipeline] step
[2022-07-07T13:53:29.534Z] --------------------------------------------------------------------------
[2022-07-07T13:53:29.534Z] Getting ORACLE_HOME...
[2022-07-07T13:53:29.534Z] Using custom SQL*Plus location
[2022-07-07T13:53:29.534Z] SQL*Plus directory: /abs/oracle/instantclient_19_9/sqlplus
[2022-07-07T13:53:29.534Z] --------------------------------------------------------------------------
[2022-07-07T13:53:29.534Z] Using custom ORACLE_HOME
[2022-07-07T13:53:29.534Z] --------------------------------------------------------------------------
[2022-07-07T13:53:29.534Z] Using ORACLE_HOME =/abs/oracle/instantclient_19_9/
[2022-07-07T13:53:29.534Z] --------------------------------------------------------------------------
[2022-07-07T13:53:29.534Z] Getting SQLPlus version
[2022-07-07T13:53:32.061Z] [ORA Sandbox] $ docker exec --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** ee6c511036173bb15b790d0c952a4cdd0ed117ea5cfe9aec13a3ea96d168bf31 /abs/oracle/instantclient_19_9/sqlplus -v
[2022-07-07T13:53:32.289Z] /abs/oracle/instantclient_19_9/sqlplus: error while loading shared libraries: libsqlplus.so: cannot open shared object file: No such file or directory
[2022-07-07T13:53:29.782Z] Process exited with status 127
[2022-07-07T13:53:29.782Z] --------------------------------------------------------------------------
[2022-07-07T13:53:29.782Z] --------------------------------------------------------------------------
[2022-07-07T13:53:29.782Z] Using ORACLE_HOME =/abs/oracle/instantclient_19_9/
[2022-07-07T13:53:29.782Z] --------------------------------------------------------------------------
[2022-07-07T13:53:29.783Z] Running defined script on ABSSOLUTE90602/********@ABSS19
[2022-07-07T13:53:29.801Z] Temp script: file:/abs/workspace/ORA%20Sandbox/temp-script-1657202009783299863286567694282.sql
[2022-07-07T13:53:29.801Z] --------------------------------------------------------------------------
[2022-07-07T13:53:32.327Z] [ORA Sandbox] $ docker exec --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** --env ******** ee6c511036173bb15b790d0c952a4cdd0ed117ea5cfe9aec13a3ea96d168bf31 /abs/oracle/instantclient_19_9/sqlplus -L ******** "@/abs/workspace/ORA Sandbox/temp-script-1657202009783299863286567694282.sql"
[2022-07-07T13:53:32.603Z] 
[2022-07-07T13:53:32.604Z] SQL*Plus: Release 19.0.0.0.0 - Production on Thu Jul 7 13:53:32 2022
[2022-07-07T13:53:32.604Z] Version 19.9.0.0.0
[2022-07-07T13:53:32.604Z] 
[2022-07-07T13:53:32.604Z] Copyright (c) 1982, 2020, Oracle.  All rights reserved.
[2022-07-07T13:53:32.604Z] 
[2022-07-07T13:53:35.250Z] ERROR:
[2022-07-07T13:53:35.250Z] ORA-12154: TNS:could not resolve the connect identifier specified
[2022-07-07T13:53:35.250Z] 
[2022-07-07T13:53:35.250Z] 
[2022-07-07T13:53:35.250Z] SP2-0751: Unable to connect to Oracle.  Exiting SQL*Plus
[2022-07-07T13:53:32.758Z] Process exited with status 1
[2022-07-07T13:53:32.762Z] --------------------------------------------------------------------------
[2022-07-07T13:53:32.762Z] Exit code : 1
[2022-07-07T13:53:32.762Z] --------------------------------------------------------------------------
[2022-07-07T13:53:32.762Z] java.lang.RuntimeException: Process exited with error
[2022-07-07T13:53:32.762Z] 	at org.jenkinsci.plugins.sqlplus.script.runner.SQLPlusRunner.run(SQLPlusRunner.java:504)
[2022-07-07T13:53:32.762Z] 	at org.jenkinsci.plugins.sqlplus.script.runner.SQLPlusRunnerBuilder.perform(SQLPlusRunnerBuilder.java:174)
[2022-07-07T13:53:32.762Z] 	at jenkins.tasks.SimpleBuildStep.perform(SimpleBuildStep.java:112)
[2022-07-07T13:53:32.762Z] 	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:116)
[2022-07-07T13:53:32.762Z] 	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:71)
[2022-07-07T13:53:32.762Z] 	at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution.lambda$start$0(SynchronousNonBlockingStepExecution.java:47)
[2022-07-07T13:53:32.762Z] 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[2022-07-07T13:53:32.762Z] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[2022-07-07T13:53:32.762Z] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[2022-07-07T13:53:32.762Z] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[2022-07-07T13:53:32.762Z] 	at java.lang.Thread.run(Thread.java:748)
[2022-07-07T13:53:32.769Z] [Pipeline] }
[2022-07-07T13:53:32.777Z] [Pipeline] // stage
[2022-07-07T13:53:32.786Z] [Pipeline] stage
[2022-07-07T13:53:32.787Z] [Pipeline] { (Declarative: Post Actions)
[2022-07-07T13:53:32.799Z] [Pipeline] echo
[2022-07-07T13:53:32.799Z] ========always========
[2022-07-07T13:53:32.812Z] [Pipeline] echo
[2022-07-07T13:53:32.813Z] ========pipeline execution failed========
[2022-07-07T13:53:32.819Z] [Pipeline] }
[2022-07-07T13:53:32.823Z] [Pipeline] // stage
[2022-07-07T13:53:32.826Z] [Pipeline] }
[2022-07-07T13:53:35.355Z] $ docker stop --time=1 ee6c511036173bb15b790d0c952a4cdd0ed117ea5cfe9aec13a3ea96d168bf31
[2022-07-07T13:53:36.723Z] $ docker rm -f ee6c511036173bb15b790d0c952a4cdd0ed117ea5cfe9aec13a3ea96d168bf31
[2022-07-07T13:53:34.277Z] [Pipeline] // withDockerContainer
[2022-07-07T13:53:34.282Z] [Pipeline] }
[2022-07-07T13:53:34.291Z] [Pipeline] // node
[2022-07-07T13:53:34.297Z] [Pipeline] End of Pipeline
[2022-07-07T13:53:34.321Z] ERROR: Process exited with error
[2022-07-07T13:53:34.321Z] Finished: FAILURE
```

**Excluding the status codes 127 - we are having those on production and it does not appear to affect the result of the pipeline so we did not bother fixing them.**

The pipeline:
```
pipeline{
    agent{
		docker {
		    label "angularBuilder"
			image "builder:1.0"
		}
    }
    options {
        skipDefaultCheckout()
    }
    stages{
        stage("Stage 1 - Checkout"){
            steps{
				step([$class: 'SQLPlusRunnerBuilder',credentialsId:"DBSchema", instance:"SID", customOracleHome:"/abs/oracle/instantclient_19_9/", customSQLPlusHome:"/abs/oracle/instantclient_19_9/sqlplus", scriptType:'userDefined', script: '', scriptContent: "select 1 from dual"])
            }
        }
    }
    post{
        always{
            echo "========always========"
        }
        success{
            echo "========pipeline executed successfully ========"
        }
        failure{
            echo "========pipeline execution failed========"
        }
    }
}
```

In the pipeline step I changed the Credentials ID and DBSchema in order to conceal the real credentials.

**Comment by boaglio** [comment]
There is a problem with your variables because SQL*PLus can't get even the version:

`
abs/oracle/instantclient_19_9/sqlplus: error while loading shared libraries: libsqlplus.so: cannot open shared object file: No such file or directory`

**Comment by raducanuoctavian** [comment]
Hello @boaglio ,

As I was saying before, we are running the scripts with this error outside of the container and it works, see the log below please!
```
18:16:40  --------------------------------------------------------------------------
18:16:40  Getting ORACLE_HOME...
18:16:40  Using custom SQL*Plus location
18:16:40  SQL*Plus directory: /abs/oracle/instantclient_19_9/sqlplus
18:16:40  --------------------------------------------------------------------------
18:16:40  Using custom ORACLE_HOME
18:16:40  --------------------------------------------------------------------------
18:16:40  Using ORACLE_HOME =/abs/oracle/instantclient_19_9
18:16:40  --------------------------------------------------------------------------
18:16:40  Getting SQLPlus version
18:16:40  [hBuilder_9.04.03.RELEASE.patches] $ /abs/oracle/instantclient_19_9/sqlplus -v
18:16:40  /abs/oracle/instantclient_19_9/sqlplus: error while loading shared libraries: libsqlplus.so: cannot open shared object file: No such file or directory
18:16:40  Process exited with status 127
18:16:40  --------------------------------------------------------------------------
18:16:40  --------------------------------------------------------------------------
18:16:40  Using ORACLE_HOME =/abs/oracle/instantclient_19_9
18:16:40  --------------------------------------------------------------------------
18:16:40  Running defined script on ****/********@ABSS19
18:16:40  Temp script: file:/abs/workspace/hBuilder_9.04.03.RELEASE.patches/temp-script-16578982006018284471757679466255.sql
18:16:40  --------------------------------------------------------------------------
18:16:40  [hBuilder_9.04.03.RELEASE.patches] $ /abs/oracle/instantclient_19_9/sqlplus -L ******** @/abs/workspace/hBuilder_9.04.03.RELEASE.patches/temp-script-16578982006018284471757679466255.sql
18:16:40  
18:16:40  SQL*Plus: Release 19.0.0.0.0 - Production on Fri Jul 15 18:16:40 2022
18:16:40  Version 19.9.0.0.0
18:16:40  
18:16:40  Copyright (c) 1982, 2020, Oracle.  All rights reserved.
18:16:40  
18:16:40  Last Successful login time: Fri Jul 15 2022 17:59:19 +03:00
18:16:40  
18:16:40  Connected to:
18:16:40  Oracle Database 19c Enterprise Edition Release 19.0.0.0.0 - Production
18:16:40  Version 19.15.0.0.0
18:16:40  
18:16:40  
18:16:40  COUNT(S.SCRIPTNAME)
18:16:40  -------------------
18:16:40  		  1
18:16:40  
18:16:40    1* SELECT COUNT (S.SCRIPTNAME) FROM SQLSCRIPT S WHERE S.SCRIPTNAME = 'Script_09.04.04.34.sql' AND S.ISSUCCESSFUL = 'Y'
18:16:40  Disconnected from Oracle Database 19c Enterprise Edition Release 19.0.0.0.0 - Production
18:16:40  Version 19.15.0.0.0
18:16:41  Process exited with status 0
18:16:41  --------------------------------------------------------------------------
```
This log is from production, as you can see we still get the status code 127 but it's ignored and the scripts still run. Inside the container, the problem is different, related to TNS:could not resolve the connect identifier specified. 

Please let me know if you need further informations on the topic.

**Comment by boaglio** [comment]
I am still waiting for your test with _tnsping_ inside your docker instance.

**Comment by raducanuoctavian** [comment]
Hello again @boaglio,

I am trying to find how tnsping inside linux is done. I might be wrong or my research is going in the wrong place. But it seems like tnsping is not available for linux. I downloaded the full linux client and the tnsping file is empty. Can you provide a way to install tnsping inside the container? The container is running Ubuntu 18.04 for reference.

Have a great day!

edit: I found out this little script written by Laurent Schneider https://laurentschneider.com/wordpress/2012/11/tnsping-and-instant-client.html which should work as an alternative for tnsping.
After trying the script the result was **ok**. Having in mind the fact that the script uses the sqlplus mechanism I can't be sure if the test is relevant or not.

**Comment by boaglio** [comment]
You can try with an official image, like oracle/database:11.2.0.2-xe

**Comment by raducanuoctavian** [comment]
After using the official client image like you suggested, I've tested a connection to our database and it worked.

![image](https://user-images.githubusercontent.com/82752011/179776581-9bddc6ab-4640-4dd4-9b7f-c4b8759dfffd.png)

Censored the host for security purpose*

tnsnames.ora file:
```
ABSS19 =
  (DESCRIPTION =
    (ADDRESS = (PROTOCOL = TCP)(HOST = server.internal.domain.com)(PORT = 1521))
    (CONNECT_DATA =
      (SERVER = DEDICATED)
      (SERVICE_NAME = ABSS19)
    )
  )
```
For this to work, had to create a tnsnames.ora file inside network/admin folder. We are using sqlplus without parsing tnsnames.ora like so:

`sqlplus -L {db_schema_name}/{db_password}@{server_ip}:{port}/{sid}`

Please let me know if you want to conduct further tests using official container images.

**Comment by boaglio** [comment]
There's nothing in the sqlplus scriptrunner plugin code that makes any difference if you're running inside a container, so I assume there's something wrong with your instance or envs. Maybe you can try to use a valid TNSNAMES.ORA file .

**Comment by raducanuoctavian** [comment]
Hi @boaglio,
Thank you for your help! After reviewing the step arguments, we saw the fact that indeed the plugin is using tnsnames.ora and it's not possible to use it without it. We did not think that tnsnames.ora is required, we tought that the plugin is working as if you were running a shell scrips as follows:
```sqlplus -L {db_schema_name}/{db_password}@{server_ip}:{port}/{sid}```
It seems like the plugin is indeed using tnsnames.ora so to fix this we updated the image adding tnsnames.ora file inside of it. 
After testing, we concluded that it was indeed a fault on our side. 

Thank you for your patiance and sorry for the missunderstanding.
Have a great day!
Octavian
